test/powershell/enginecore/BooleanParameterDCR.Tests.ps1

```diff
@@ -0,0 +1,52 @@
+Describe "BooleanParameterDCR Tests" -tags "CI" {
+    BeforeAll {
+        Function ParserTestSwitchCmdlet
+        {
+          [CmdletBinding()]
+          param([switch]$switchParam)
+          return $switchParam
+        }
+        
+        Function ParserTestBoolCmdlet2
+        {
+          [CmdletBinding()]
+          param([bool]$First=$false)
+          return $First
+        }
+    }
+    
+    $tests = @(
+            @{ inputTest = 0; expected = $false },
+            @{ inputTest = 000; expected = $false },
+            @{ inputTest = 0x00; expected = $false },
+            @{ inputTest = 0.00; expected = $false }
+    )
+    It "Test that passing zero works as the value for a Switch parameter, inputTest:<inputTest>,expect:<expected>" -TestCases $tests {
+            param ( $inputTest, $expected )
+            [bool]$switchTestParam = $inputTest
+            $result = ParserTestSwitchCmdlet -switchParam:$switchTestParam
+            $result | should be $expected
+    }
+    
+    $tests = @(
+            @{ inputTest = $(1 -eq 1); expected = $true },
+            @{ inputTest = $true; expected = $true },
+            @{ inputTest = $TRUE; expected = $true }

```

Comment:
I'm not sure whether this is needed - is the validation trying to determine the difference between $true and $TRUE